### DISCUSSION TITLE
Installing in non-Anaconda-based Jupyterlab instance

### DISCUSSION THREAD

**Discussion by samrblackman** [jupyterlab/jupyterlab-git#1263] [Q&A]
Hi Team -

I'm trying to get this working in a Jupyterlab instance I've created without Anaconda and can't seem to get it working. I've set my Jupyterlab version < 4 and installed Git and node.js (manually from their site), yet no dice. I keep getting "git command not found make sure Git version >2 is installed" messages. 

I also have installed the lab extension and it shows up as "OK" and so I'm not sure what is the issue. I've read that this cannot work in a Jupyterlab instance created outside of Anaconda (I just built my own virtual environment).

Sorry if this is an ignorant question; please point me in the right direction if so! Just not sure what to do at this point. Thanks!

**Comment by fcollonval** [comment]
Hey @samrblackman 

This extension can definitely be installed in non-Anaconda-based instance. You should check:
- git command is available in whatever environment isolation you are using
- node.js is not needed - installing through pip with `pip install jupyterlab-git` will give you the JavaScript assets directly

To receive more help, you will need to share more info about your system. In particular, you should execute JupyterLab in debug mode (using the `--debug` option).

> of importance, don't install Python package in the user space (using `pip install --user`) it is known to behave poorly with JupyterLab extension.

**Comment by samrblackman** [comment]
Thanks Frederic,

Perhaps the below details will help.

I have installed git and checked the option for my system path variables to
be updated. This appears to install successfully.

I then run the following within a .bat script (as admin) to create my
jupyterlab instance. <USER> is a placeholder for my user profile name.

cd C:\Users\<USER>\JupyterLab\workspace
C:
SET JUPYTER_RUNTIME_DIR=C:\Users\ <USER>  \JupyterLab\runtime
SET JUPTER_DATA_DIR=C:\Users\ <USER>  \JupyterLab\jupyter
SET SHELL=C:\Windows\SysWOW64\WindowsPowerShell\v1.0\powershell.exe
SET PATH=C:\Program Files\Python311;C:\Program
Files\Python311\Scripts;C:\Program Files\Python311\Lib;C:\Program
Files\Python311\include
call python -m venv C:\Users\ <USER>  \JupyterLab\gitenv
call C:\Users\ <USER>  \JupyterLab\gitenv\Scripts\activate.bat

call python -m pip install --upgrade pip
call pip install pywinpty
call pip install requests
call pip install jupyter
call pip install -v "jupyterlab<4"
call pip install jupyterlab-git
call ipython kernel install
call pip install pandas

call jupyter lab --debug

This seems to run successfully, and then when the JupyterLab instance
launches in my Chrome browser, I get the error message "Failed to load the
jupyterlab-git server extension. Git command not found - please ensure you
have Git > 2 installed."

I don't understand as I have Git installed and as far as I know the correct
path variables are set up. I can access git via cmd and powershell outside
of jupyterlab, so I am confused!

Thanks for your help and please let me know if there is anything else I can
do to help.

**Reply by fcollonval** [reply]
Would you mind execute in a notebook cell

```py
import os
from shutil import which
print(which('git'))
```

and report here what is the result

**Comment by samrblackman** [comment]
Sure thing - the result is "None" which doesn't seem good! Thanks for your
help.

---------------------
Sam Blackman
1-207-233-8943
***@***.***
https://www.linkedin.com/in/samrblackman/

**Comment by samrblackman** [comment]
Hi Frederic, I'm sure you're busy, I thought I'd ping you. Have you had a
chance to look into this further? Thanks for your help.

-Sam

**Comment by samrblackman** [comment]
Hi Frederic, any updates? Appreciate your help. Thanks

**Comment by samrblackman** [comment]
Hi team,

It's been 5 months or so, and I still have this issue. Any direction would
be appreciated. Thanks... I'd like to use Git in my jupyterlab yet this
seems unfeasible.

Thanks
Sam
---------------------
Sam Blackman
1-207-233-8943
***@***.***
